compute/instance-templates/create-instance-templates/create_instance_templates_test.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestCreateInstanceTemplatesSnippets(t *testing.T) {
+	var seededRand *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "europe-central2-b"
+	instanceName := "test-instance-" + fmt.Sprint(seededRand.Int())
+	templateName1 := "test-template-" + fmt.Sprint(seededRand.Int())
+	templateName2 := "test-template-" + fmt.Sprint(seededRand.Int())
+	templateName3 := "test-template-" + fmt.Sprint(seededRand.Int())
+	machineType := "n1-standard-1"
+	sourceImage := "projects/debian-cloud/global/images/family/debian-10"
+	networkName := "global/networks/default"
+	subnetworkName := "regions/asia-east1/subnetworks/default"
+
+	ctx := context.Background()
+
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Errorf("NewInstancesRESTClient: %v", err)
```

Comment:
nit: I think many(or all?) of these should be t.Fatalf. This test in others seem sequential so it does not make sense to keep running the test if you can't create the client for instance.

Reply:
For clients creating agreed.